pkg/archive/jar.go

```diff
@@ -105,3 +105,25 @@ var StableJAROrderOfAttributeValues = ZipEntryStabilizer{
 		zf.SetContent(buf.Bytes())
 	},
 }
+
+var StableGitProperties = ZipArchiveStabilizer{
+	Name: "jar-git-properties",
+	Func: func(mr *MutableZipReader) {
+		for _, mf := range mr.File {
+			// These files contain git properties set by git-commit-id-maven-plugin.
+			// They contain many unreproducible attributes as documented
+			// [here](https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/825).
+
+			// Only JSON and properties file formats are available as documented
+			// [here](https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/95d616fc7e16018deff3f17e2d03a4b217e55294/src/main/java/pl/project13/maven/git/GitCommitIdMojo.java#L454).
+			// By default, these file are created in ${project.build.outputDirectory} and are hence at the root of the jar.
+			// We assume that the file name is 'git' as this is the default value for the plugin.
+			// However, the plugin allows customizing the file name.
+			if strings.HasSuffix(mf.Name, "git.properties") || strings.HasSuffix(mf.Name, "git.json") {
```

Comment:
do we want to add a / separator here to avoid accidentally omitting a file like `plugit.json` accidentally?

Reply:
I instead added a word boundary `\b` as the default location in the jar file is `git.json` and a separator would not match there, but a word boundary will match both.

Reply:
yeah I think this comes down to the discussion in the other thread so let's close this one out.